### PR TITLE
ROX-9497: Make use of the injected CA bundle

### PR DESCRIPTION
**Pull request by 0x656b694d**
Update the `import-additional-cas` to copy the injected CA bundle to the common location before calling `update-ca-trust`.

> Cluster administrators can configure additional trusted CAs in OpenShift. The combined set of trusted CAs (from a cluster perspective) can be obtained by creating an empty ConfigMap with a special label attached to it. An operator will then inject the trusted CAs into a well-defined field of the configmap.

Related PR to stackrox: https://github.com/stackrox/stackrox/pull/174

### CODE REVIEW

File: image/scanner/scripts/import-additional-cas [hunk 1]
```diff
@@ -1,8 +1,17 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 set -euo pipefail
 
-[ -d /usr/local/share/ca-certificates ] || exit 0
-[ "$(find /usr/local/share/ca-certificates -maxdepth 1 -name '*.crt' | wc -l)" -gt 0 ] || exit 0
-cp -L /usr/local/share/ca-certificates/* /etc/pki/ca-trust/source/anchors
+copy_existing () {
+    src=$1
```
Comment (msugakov) [inline image/scanner/scripts/import-additional-cas:6]: [nit]
```suggestion
    src="$1"
```

File: image/scanner/scripts/import-additional-cas [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+    if [ -d "$src" ] && [ "$(ls -A -I "..*" "$src")" ]; then
+        cp -v -L "$src"/* /etc/pki/ca-trust/source/anchors
+    else
+        echo "No certificates found in $src"
```
Comment (msugakov) [inline image/scanner/scripts/import-additional-cas:10]: Maybe just skip it quietly?
```suggestion
```

File: image/scanner/scripts/import-additional-cas [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+    fi
+}
+
+copy_existing /usr/local/share/ca-certificates
+copy_existing /etc/pki/injected-ca-trust
+
 update-ca-trust
```
Comment (msugakov) [inline image/scanner/scripts/import-additional-cas:17]:
```suggestion
update-ca-trust extract
```

File: image/scanner/scripts/import-additional-cas [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
 
 set -euo pipefail
 
-[ -d /usr/local/share/ca-certificates ] || exit 0
-[ "$(find /usr/local/share/ca-certificates -maxdepth 1 -name '*.crt' | wc -l)" -gt 0 ] || exit 0
-cp -L /usr/local/share/ca-certificates/* /etc/pki/ca-trust/source/anchors
+copy_existing () {
+    src=$1
+    if [ -d "$src" ] && [ "$(ls -A -I "..*" "$src")" ]; then
+        cp -v -L "$src"/* /etc/pki/ca-trust/source/anchors
```
Comment (msugakov) [inline image/scanner/scripts/import-additional-cas:8]: On one hand I like `-v`, on the other hand, this code is called each time the image is started and will produce some logging that may be noisy. WDYT?

File: image/scanner/scripts/import-additional-cas [hunk 5]
```diff
@@ -1,8 +1,17 @@
...
+#!/usr/bin/env bash
 
 set -euo pipefail
 
-[ -d /usr/local/share/ca-certificates ] || exit 0
-[ "$(find /usr/local/share/ca-certificates -maxdepth 1 -name '*.crt' | wc -l)" -gt 0 ] || exit 0
-cp -L /usr/local/share/ca-certificates/* /etc/pki/ca-trust/source/anchors
+copy_existing () {
+    src=$1
+    if [ -d "$src" ] && [ "$(ls -A -I "..*" "$src")" ]; then
```
Comment (msugakov) [inline image/scanner/scripts/import-additional-cas:7]: 1. Why is `-I "..*"` needed?
2. You can merge predicates
```suggestion
if [[ -d "$src" && -n "$(ls -A -I "..*" "$src")" ]]; then
```

File: image/scanner/scripts/import-additional-cas (hunk 2 above)
Comment (msugakov) [reply image/scanner/scripts/import-additional-cas:10]: Feel free to ignore for consistency with https://github.com/stackrox/stackrox/pull/1088

File: image/scanner/scripts/import-additional-cas (hunk 4 above)
Comment (msugakov) [reply image/scanner/scripts/import-additional-cas:8]: Feel free to ignore for consistency with https://github.com/stackrox/stackrox/pull/1088

Comment (0x656b694d) [reply image/scanner/scripts/import-additional-cas:8]: That `-v` was added specifically for generating the little noise for logging purposes to simplify debugging.

File: image/scanner/scripts/import-additional-cas (hunk 5 above)
Comment (0x656b694d) [reply image/scanner/scripts/import-additional-cas:7]: 1. `-I "..*"` is there to ignore files like "..data" created by kubernetes volume mounting.
2. Yes, but why?

Comment (msugakov) [reply image/scanner/scripts/import-additional-cas:7]: 2: Clarity. Each `[` spawns a command. In `[ something ] && [ something ]` there are at least two `[` commands.
Unlike `[`, `[[` is, I believe, a bash built-in that doesn't span extra command on its own. Plus `&&` inside of `[` or `[[` doesn't cause another command execution.
By _command_ I mean starting a process.

Comment (0x656b694d) [reply image/scanner/scripts/import-additional-cas:7]: Hm. I'm not sure how `[[a && -n b]]` is more clear than `[a] && [b]`.

Comment (msugakov) [reply image/scanner/scripts/import-additional-cas:7]: You can try `[[ a && b ]]`. Most likely it will work the same. Will it still be not clear enough?

Comment (msugakov) [reply image/scanner/scripts/import-additional-cas:7]: Regarding
> `-I "..*"` is there to ignore files like "..data" created by kubernetes volume mounting.

It is really not obvious from just looking at the code. I would recommend you adding this info as a comment in the code.

**Comment by 0x656b694d** [comment]
Merged with #692

**Pull request closed without merging**